corehq/apps/user_importer/tasks.py

```diff
@@ -37,10 +42,50 @@ def _update_progress(value, start=0):
         'rows': user_results['rows']
     }
     upload_record = UserUploadRecord.objects.using(DEFAULT_DB_ALIAS).get(pk=upload_record_id)
-    upload_record.task_id = import_users_and_groups.request.id
+    upload_record.task_id = task.request.id
     upload_record.result = results
     upload_record.save()
     DownloadBase.set_progress(task, total, total)
     return {
         'messages': results
     }
+
+
+@task(serializer='pickle', queue='user_import_queue')
+def parallel_import_task(domain, user_specs, group_specs, upload_user, upload_record_id):
+    task = parallel_import_task
+    return import_users_and_groups(domain, user_specs, group_specs, upload_user, upload_record_id, task)
+
+
+@task(serializer='pickle', queue='user_import_queue')
+def parallel_user_import(domain, user_specs, group_specs, upload_user):
+    task = parallel_user_import
+    total = len(user_specs) + len(group_specs)
+    DownloadBase.set_progress(task, 0, total)
+    task_list = []
+    for users in chunked(user_specs, 100):
+        upload_record = UserUploadRecord(
+            domain=domain,
+            user_id=upload_user.user_id
+        )
+        upload_record.save()
+
+        subtask = parallel_import_task.delay(
+            domain,
+            list(users),
+            list(group_specs),
+            upload_user,
+            upload_record.pk
+        )
+        task_list.append(subtask)
+    incomplete = True
+    while incomplete:
```

Comment:
It is. I don't love the idea of sleeping because it already takes a long time to update the progress bar (1 second per task, which is already a couple minutes for the real event sizing), but I could do that if you are concerned about load.